docs/api/app.md

```diff
@@ -228,6 +228,10 @@ app.on('login', function(event, webContents, request, authInfo, callback) {
 
 Emitted when the gpu process crashes.
 
+### Event: 'dark-mode-changed' _OS X_
```

Comment:
Sounds good to me :+1: 
I'll go with `platform-theme-changed` as it might be clearer for when the event is not OSX specific.

Also, I'm not too versed in coding for Windows so I won't try to add that functionality in this PR